app/src/pages/inside/projectSettingsPageContainer/projectSettingsPageContainer.jsx

```diff
@@ -14,39 +14,161 @@
   });
 
-  render() {
-    return (
-      <PageLayout>
-        <PageHeader breadcrumbs={this.breadcrumbs} />
-        <SettingsPage projectId={this.props.projectId} createTabLink={this.createTabLink} />
-      </PageLayout>
+  const createExtensionTabs = () => {
+    return extensions.reduce(
+      (acc, extension) => ({
+        ...acc,
+        [extension.name]: {
+          name: extension.title || extension.name,
+          link: createTabLink(extension.name),
+          component: <ExtensionLoader extension={extension} />,
+          mobileDisabled: true,
+          eventInfo: SETTINGS_PAGE_EVENTS.extensionTabClick(extension.title || extension.name),
+        },
+      }),
+      {},
     );
-  }
-}
+  };
+
+  const createItemsConfig = () => {
+    const extensionTabs = createExtensionTabs();
+    const tabsConfig = {
+      [GENERAL]: {
+        name: formatMessage(messages.general),
+        link: createTabLink(GENERAL),
+        component: <GeneralTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.GENERAL_TAB,
+        mobileDisabled: true,
+      },
+      [INTEGRATIONS]: {
+        name: formatMessage(messages.integrations),
+        link: createTabLink(INTEGRATIONS),
+        component: <IntegrationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.INTEGRATIONS_TAB,
+      },
+      [NOTIFICATIONS]: {
+        name: formatMessage(messages.notifications),
+        link: createTabLink(NOTIFICATIONS),
+        component: <NotificationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.NOTIFICATIONS_TAB,
+        mobileDisabled: true,
+      },
+      [DEFECT]: {
+        name: formatMessage(messages.defect),
+        link: createTabLink(DEFECT),
+        component: <DefectTypesTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEFECT_TYPE_TAB,
+        mobileDisabled: true,
+      },
+      [ANALYSIS]: {
+        name: formatMessage(messages.analysis),
+        link: createTabLink(ANALYSIS),
+        component: <AutoAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.AUTO_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [PATTERN_ANALYSIS]: {
+        name: formatMessage(messages.patternAnalysis),
+        link: createTabLink(PATTERN_ANALYSIS),
+        component: <PatternAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.PATTERN_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [DEMO_DATA]: {
+        name: formatMessage(messages.demoData),
+        link: createTabLink(DEMO_DATA),
+        component: <DemoDataTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEMO_DATA_TAB,
+        mobileDisabled: true,
+      },
+    };
+    if (!canSeeDemoData(accountRole, userRole)) {
+      delete tabsConfig[DEMO_DATA];
+    }
+    Object.keys(extensionTabs).forEach((tab) => {
+      if (tabsConfig[tab]) {
+        tabsConfig[tab].component = extensionTabs[tab].component;
+
+        delete extensionTabs[tab];
+      }
+    });
+    return { ...tabsConfig, ...extensionTabs };
+  };
```

Comment:
Why not use an array with objects instead of an object with objects?
```
{
  id: DEMO_DATA,
  name: formatMessage(messages.demoData),
  link: createTabLink(DEMO_DATA),
  component: <DemoDataTab />,
  eventInfo: SETTINGS_PAGE_EVENTS.DEMO_DATA_TAB,
  mobileDisabled: true,
}
```
This will allow us to do something like
```
items.map(({ id, link, eventInfo, name }) => (
  <NavLink to={link} 
    key={id}
    onClick={createTrackingFunction(eventInfo)}
    className={cx('item')}
    activeClassName={cx('active')}
  >
    {name}
  </NavLink>
))
```
instead
```
Object.keys(items).map((item) => (
  <NavLink
    to={items[item].link}
    key={item}
    onClick={createTrackingFunction(items[item].eventInfo)}
    className={cx('item')}
    activeClassName={cx('active')}
  >
    {items[item].name}
  </NavLink>
))
```

Reply:
This is good advice. Unfortunately not suitable for this case. Because we manage the menu item and content with routing. It's better to be able to find an element directly with a single operation like `config[activeTab]` than to use an array and iterate over it `.find`